.konflux/scripts/render_snapshot.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+import datetime
+import json
+import os
+import re
+
+
+def parse_image_refs(image_refs):
+    return json.loads(image_refs)
+
+
+def validate_component(component):
+    assert (
+        component["name"] != ""
+        and component["containerImage"] != ""
+        and component["revision"] != ""
+        and component["repository"] != ""
+    ), "Component must have component name, ref, revision and repository set. Check container image labels."
+
+
+def determine_component_version_suffix(application):
+    match = re.search(r"acs-(?P<version>\d+-\d+)", application)
+    if match:
+        return match.group('version')
+    return ""
+
+
+def process_component(component, name_suffix):
+    validate_component(component)
+    if name_suffix != "":
+        name = f"{component['name']}{name_suffix}"
+    else:
+        name = component["name"]
+
+    return {
+        "containerImage": component["containerImage"],
+        "name": name,
+        "source": {
+            "git": {
+                "revision": component["revision"],
+                "url": component["repository"]
+            }
+        }
+    }
+
+
+def construct_snapshot(
+    snapshot_name_prefix,
+    snapshot_version_suffix,
+    pipeline_run_name,
+    namespace,
+    application,
+    components
+):
+    timestamp = datetime.datetime.now(datetime.timezone.utc).strftime("%Y%m%dT%H%M%SZ")
+    snapshot_name = f"{snapshot_name_prefix}-{snapshot_version_suffix}-{timestamp}".lower()
```

Comment:
I think, the code will become cleaner if you create a dedicated function which creates the snapshot name, call it from "main" and only pass the final `snapshot_name` into `construct_snapshot()`.